tracer/src/Datadog.Trace/AppSec/ObjectExtractor.cs

```diff
@@ -161,23 +158,33 @@ private static string GetPropertyName(string fieldName)
 
         private static object ExtractType(Type itemType, object value, int depth, HashSet<object> visited)
         {
+            if (itemType == typeof(string))
+            {
+                return value;
+            }
+
             if (itemType.IsArray || (itemType.IsGenericType && itemType.GetGenericTypeDefinition() == typeof(List<>)))
             {
                 return ExtractListOrArray(value, depth, visited);
             }
-            else if (itemType.IsGenericType && itemType.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+
+            if (itemType.IsGenericType && itemType.GetGenericTypeDefinition() == typeof(Dictionary<,>))
             {
                 return ExtractDictionary(value, itemType, depth, visited);
             }
-            else if (IsOurKindOfPrimitive(itemType))
+
+            if (WafProcessableTypes.Contains(itemType))
             {
-                return value?.ToString();
+                return value;
             }
-            else
+
+            if (itemType.IsEnum || itemType == typeof(Guid) || itemType == typeof(DateTime) || itemType == typeof(DateTimeOffset) || itemType == typeof(TimeSpan) || itemType.IsPrimitive)
```

Comment:
Maybe you want to add `typeof(DateOnly)` and `typeof(TimeOnly)` for .NET 6+? 🤷‍♂️